backend/danswer/connectors/sharepoint/connector.py

```diff
@@ -147,48 +135,48 @@ def get_all_driveitem_objects(
 
         return driveitem_list
 
-    def get_all_site_objects(self) -> list[Site]:
+    def _get_all_site_objects(self) -> list[SiteData]:
         if self.graph_client is None:
             raise ConnectorMissingCredentialError("Sharepoint")
 
         if self.site_data:
             for element in self.site_data:
-                element.siteobjects = [self.graph_client.sites.get_by_url(element.url).get().execute_query()]
+                element.siteobjects = [
+                    self.graph_client.sites.get_by_url(element.url)
+                    .get()
+                    .execute_query()
+                ]
+
+            return self.site_data
+
         else:
             site_objects = self.graph_client.sites.get().execute_query()
-            self.site_data = [SiteData(url=None, folder=None, siteobjects=site_objects, driveitems=[])]
+            return [
+                SiteData(url=None, folder=None, siteobjects=site_objects, driveitems=[])
```

Comment:
This data object has attributes that just aren't used. Not sure why they're included or maybe I just didnt read hard enough